### PR TITLE
DPDTimeout Multiple Radius for VPN Gateways (Network)

### PR DESCRIPTION
**Pull request by henry416**
Adding DPDTimeout and Multiple Radius Servers for VPN Gateways
DPDTimeout is an integer field in connection resources.
Multiple Radius is a subobject containing radius sever address/score/secret in gateway resources.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by henry416** [comment]
https://github.com/Azure/azure-rest-api-specs/pull/8697 Was told to create PR targeting this branch

**Comment by henry416** [comment]
@lirenhe / @zikalino Can you please review this PR?

**Comment by henry416** [comment]
@number213 Can I get a review of this PR?

**Review by anton-evseev — APPROVED** [review]

**Comment by henry416** [comment]
@MikhailTryakhov / @v-shhatt Please review this PR, thanks!

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualWan.json [hunk 1]
```diff
@@ -3623,6 +3623,11 @@
           "format": "int32",
           "description": "Routing weight for vpn connection."
         },
+        "dpdTimeoutSeconds": {
+          "type": "integer",
+          "format": "int32",
+          "description": "DPD timeout for a vpn connection in seconds."
```
Comment (pilor) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualWan.json]: >DPD [](start = 26, length = 3)

Spelling this out in the description would be more helpful to users

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json [hunk 2]
```diff
@@ -2459,6 +2459,24 @@
...
+        "radiusServerAddress": {
+          "type": "string",
+          "description": "The address of this radius server."
+        },
+        "radiusServerScore": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The initial score assigned to this radius server."
+        },
+        "radiusServerSecret": {
```
Comment (pilor) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json]: >"radiusServerSecret" [](start = 8, length = 20)

Secrets should not be exposed in GETs or as the responses to PUT/PATCH.  They should only be exposed in POST calls (but can be set via PUT/PATCH as long as they don't show up in the response).  How is this secret handled in the API?

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json [hunk 3]
```diff
@@ -2459,6 +2459,24 @@
       ],
       "description": "VPN client revoked certificate of virtual network gateway."
     },
+    "RadiusServer": {
```
Comment (pilor) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json]: >"RadiusServer" [](start = 4, length = 14)

please add at least one example that contains radius servers to all APIs that are affected by this change

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json [hunk 4]
```diff
@@ -2459,6 +2459,24 @@
...
+      "properties": {
+        "radiusServerAddress": {
```
Comment (pilor) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json]: >"radiusServerAddress" [](start = 8, length = 21)

if any of these properties are required when specifying a radius server please mark them as such

**Review by pilor — CHANGES_REQUESTED** [review]
:clock1:

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json (hunk 4 above)
Comment (henry416) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json]: Done for radius address

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json (hunk 3 above)
Comment (henry416) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json]: Done.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json (hunk 2 above)
Comment (henry416) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json]: This secret is not exposed. Only set through put

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualWan.json (hunk 1 above)
Comment (henry416) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualWan.json]: Done. DPD = Dead Peer Detection

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VpnServerConfigurationList.json [hunk 5]
```diff
@@ -105,8 +105,13 @@
                   "thumbprint": "83FFBFC8848B5A5836C94D0112367E16148A286F"
                 }
               ],
-              "radiusServerAddress": "8.9.9.9",
-              "radiusServerSecret": "123_abc",
+              "vpnClientRevokedCertificates": [
```
Comment (pilor) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VpnServerConfigurationList.json]: >"vpnClientRevokedCertificates" [](start = 14, length = 30)

copy paste error?

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json (hunk 2 above)
Comment (pilor) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json]: The examples should not return the property at all, right now they show an empty string.  Also, please add x-ms-mutability to make it clear to consumers when/where this property will show up  https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-mutability

**Review by pilor — CHANGES_REQUESTED** [review]
1 previous comment still open and 1 minor new comment

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VpnServerConfigurationList.json (hunk 5 above)
Comment (henry416) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VpnServerConfigurationList.json]: Fixed

**Comment by henry416** [comment]
@pilor Comments addressed in latest commits

**Review by pilor — APPROVED** [review]

**Comment by number213** [comment]
@henry416 please resolve CI errors

**Comment by henry416** [comment]
@number213 I have fixed the CI errors in examples jsons. The remaining checks don't display any actionable error that I can find. My local autorest validation doesn't show any error for the changes I made.

**Comment by henry416** [comment]
@number213 could you please advise on what the CI errors are? I'm a bit stuck. The pipelines don't show a descriptive error, and autorest/linter validations don't show anything off.

**Comment by number213** [comment]
@henry416 PrettierCheck is throwing errors due to JSON files not being prettified correctly

```
prettier -c specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VirtualNetworkGatewayList.json
Checking formatting...
specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VirtualNetworkGatewayList.json
Code style issues found in the above file(s). Forgot to run Prettier?

prettier -c specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VirtualNetworkGatewayUpdate.json
Checking formatting...
specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VirtualNetworkGatewayUpdate.json
Code style issues found in the above file(s). Forgot to run Prettier?

prettier -c specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VpnServerConfigurationGet.json
Checking formatting...
specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VpnServerConfigurationGet.json
Code style issues found in the above file(s). Forgot to run Prettier?

prettier -c specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VpnServerConfigurationPut.json
Checking formatting...
specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/examples/VpnServerConfigurationPut.json
Code style issues found in the above file(s). Forgot to run Prettier?

prettier -c specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json
Checking formatting...
specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetworkGateway.json
Code style issues found in the above file(s). Forgot to run Prettier?

```

**Comment by henry416** [comment]
@number213 I have fixed the prettier output. Is there any way I can run the prettier check on files in local? I cannot find documentation in the repo on how to run it.
Looks like the remaining checks are about breaking changes, which should be okay since the new fields added are optional fields and do not alter existing scenarios.

**Comment by henry416** [comment]
@number213 Ping for visibility

**Comment by number213** [comment]
@zikalino @akning-ms could you please merge this PR? Breaking change job can be ignored since this is a PR for Network's dev branch

@henry416 you can run prettier locally with `npm run-script prettier-check`

**Pull request merged**